zotero/src/data_structure/item/mod.rs

```diff
@@ -204,17 +497,87 @@ pub struct Creator {
     pub last_name: String,
 }
 
+impl Creator {
+    pub fn full_name(&self) -> String {
+        format!("{} {}", self.first_name, self.last_name)
+    }
+
+    pub fn short_name(&self) -> String {
+        if self.first_name.len() == 0 {
+            return format!("{}", self.last_name)
+        }
+        format!("{}. {}", self.first_name.chars().next().unwrap(), self.last_name)
+    }
+}
+
 #[derive(Deserialize, Serialize, Debug, PartialEq, Clone)]
 #[serde(rename_all(deserialize = "camelCase", serialize = "camelCase"))]
 pub struct ItemMeta {
     pub creator_summary: Option<String>,
     pub parsed_date: Option<String>,
-    pub num_children: Option<usize>,
+    #[serde(deserialize_with = "deserialize_meta_num_children", default)]
+    pub num_children: Option<SizeOrBool>,
     // The following part concerns collections
     pub num_collections: Option<usize>,
     pub num_items: Option<usize>,
 }
 
+impl ItemMeta {
+    pub fn has_children(&self) -> bool {
+        match &self.num_children {
+            None => false,
+            Some(sob) => {
+                match sob {
+                    SizeOrBool::Bool(_) => false,
+                    SizeOrBool::Size(v) => {
+                        if *v > 0 {
+                            true
+                        } else {
+                            false
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub enum SizeOrBool {
```

Comment:
```suggestion
#[serde(untagged)]
pub enum SizeOrBool {
```

With `#[serde(untagged)]` serde will try to deserialize against each variant.